### PR TITLE
Add some examples

### PR DESCRIPTION
**Pull request by zainab-ali**
Adding examples as referenced in #86

### CODE REVIEW

File: tests/shared/src/test/scala/matryoshka/example/intList.scala [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    (0 until 10).toList.hylo(filter(_ < 5), to) should ===((0 until 5).toList)
+  }
+
+  "map the head of an IntList" >> {
+    intList.apo.apply(mapHead(_ * 3)).cata(from) should ===(List(3, 2))
+  }
+
+  // "short circuit the creation of infinite IntList" >> {
+  //   1.postpro[Fix[IntList]](IntList.lessThan(10), infinite).cata(len) should ===(9)
+  // }
```
Comment (zainab-ali) [inline tests/shared/src/test/scala/matryoshka/example/intList.scala]: I would expect this to cut the infinite stream short at 9, but it stack overflows instead.  Am I using `postpro` wrongly?

**Comment by zainab-ali** [comment]
Any updates on this?

File: tests/shared/src/test/scala/matryoshka/example/intList.scala [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  def filter(f: Int => Boolean): Algebra[IntList, List[Int]] = {
+    case IntNil() => Nil
+    case IntCons(h, t) => if(f(h)) h :: t else t
+  }
+
+  def lessThan(i: Int): IntList ~> IntList = new (IntList ~> IntList) {
+    def apply[A](l: IntList[A]): IntList[A] = l match {
+      case IntNil() =>
+        println(s"intnil")
```
Comment (jdegoes) [inline tests/shared/src/test/scala/matryoshka/example/intList.scala]: Debug code.

File: tests/shared/src/test/scala/matryoshka/example/intList.scala [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        IntNil()
+      case l @ IntCons(h, t) =>
+        println(s"head is $h")
```
Comment (jdegoes) [inline tests/shared/src/test/scala/matryoshka/example/intList.scala]: Debug code.

File: tests/shared/src/test/scala/matryoshka/example/expr.scala [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package matryoshka.example
+
```
Comment (jdegoes) [inline tests/shared/src/test/scala/matryoshka/example/expr.scala]: Extraneous newline (here and elsewhere).

**Comment by zainab-ali** [comment]
Thanks for spotting the `println` statements!  I'm still trying to get my head around the `postpro` for `IntList`, but will keep investigating.

**Pull request merged**